models/spring-ai-vertex-ai-gemini/src/main/java/org/springframework/ai/vertexai/gemini/VertexAiGeminiChatClient.java

```diff
@@ -243,8 +235,19 @@ private GeminiRequest createGeminiRequest(Prompt prompt) {
 		}
 
 		// Add the enabled functions definitions to the request's tools parameter.
+
+		List<Tool> tools = new ArrayList<>();
 		if (!CollectionUtils.isEmpty(functionsForThisRequest)) {
-			List<Tool> tools = this.getFunctionTools(functionsForThisRequest);
+			tools.addAll(this.getFunctionTools(functionsForThisRequest));
+		}
+		if (((VertexAiGeminiChatOptions) prompt.getOptions()).getGoogleSearchRetrieval()) {
```

Comment:
There is a NPE here when running the tests.   I have fixed it, but it means this code wasn't really verified.

In future, please add a test to verify the behavior.